libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx

```diff
@@ -0,0 +1,299 @@
+        service.enum !== null && !service.id && !service.markedForDeletion
+    );
+
+    const toRemoveOtherServices = serviceOthers.filter(
+      (service) => service.markedForDeletion && !!service.id
+    );
+
+    const toRemoveServices = services.filter(
+      (service) => service.markedForDeletion && !!service.id
+    );
+
+    const toRemoveServicesWithOther = [
+      ...toRemoveOtherServices,
+      ...toRemoveServices,
+    ];
+
+    try {
+      for (const service of toRemoveServicesWithOther) {
+        if (service.id) {
+          try {
+            await deleteService({
+              variables: {
+                data: {
+                  id: service.id,
+                },
+              },
+            });
+          } catch (error) {
+            console.error(
+              `Error deleting service with id ${service.id}:`,
+              error
+            );
+          }
+        }
+      }
+
+      for (const service of toCreateServices) {
+        if (service.enum) {
+          try {
+            await createService({
+              variables: {
+                data: {
+                  service: service.enum,
+                  noteId,
+                  serviceRequestType: type,
+                },
+              },
+            });
+          } catch (error) {
+            console.error(
+              `Error creating service with enum ${service.enum}:`,
+              error
+            );
+          }
+        }
+      }
+
+      for (const service of toCreateOtherServices) {
+        if (service.title) {
+          try {
+            await createService({
+              variables: {
+                data: {
+                  service: ServiceEnum.Other,
+                  serviceOther: service.title,
+                  noteId,
+                  serviceRequestType: type,
+                },
+              },
+            });
+          } catch (error) {
+            console.error(
+              `Error creating service with title ${service.title}:`,
+              error
+            );
+          }
+        }
+      }
+
+      refetch();
+      setIsModalVisible(false);
+    } catch (e) {
+      console.error('Error during service submission:', e);
+    }
+  };
+
+  const closeModal = () => {
+    const newInitialServices = initialServices
+      .filter((item) => item.service !== ServiceEnum.Other)
+      .map((service) => ({ id: service.id, enum: service.service }));
+    const initialServiceOthers = initialServices
+      .filter((item) => item.service === ServiceEnum.Other)
+      .map((service) => ({
+        id: service.id,
+        title: service.serviceOther || '',
+      }));
+    setIsModalVisible(false);
+    setServiceOthers(initialServiceOthers);
+    setServices(newInitialServices);
+  };
+
+  useEffect(() => {
+    const newInitialServices = initialServices
+      .filter((item) => item.service !== ServiceEnum.Other)
+      .map((service) => ({ id: service.id, enum: service.service }));
```

Comment:
to be consistent I'd suggest user either `.filter(item) and .map(item)` in both or `service` in both as they're same objects

Reply:
renamed to serviceItem both